ocs_ci/templates/workloads/helper_scripts/meta_data_io.py

```diff
@@ -0,0 +1,111 @@
+    try:
+        file_paths = []
+        for i in range(1, num_files + 1):
+            file_path = os.path.join(base_path, f"file{i}.txt")
+            with open(file_path, "w") as f:
+                f.write(f"This is file {i}")
+                pass
+            file_paths.append(file_path)
+        return file_paths
+    except Exception as e:
+        logging.error(f"Error during file creation: {str(e)}")
+        return []
+
+
+def get_extended_attribute(file_path, attr_name):
+    try:
+        attr_value = os.getxattr(file_path, attr_name)
+        return attr_value.decode("utf-8")
+    except (OSError, IOError) as e:
+        logging.error(f"Error getting extended attribute: {str(e)}")
+        return None
+
+
+def set_extended_attribute(file_path, attr_name, attr_value):
+    try:
+        os.setxattr(file_path, attr_name, attr_value.encode("utf-8"))
+    except (OSError, IOError) as e:
+        logging.error(f"Error setting extended attribute: {str(e)}")
+
+
+def perform_metadata_operations(file_path):
+    try:
+        # Acquire a lock on the file
+        with open(file_path, "a") as lock_file:
+            fcntl.flock(lock_file, fcntl.LOCK_EX)
+
+            # Get initial metadata
+            initial_metadata = os.stat(file_path)
+            log_metadata_operation("Initial", file_path, None, initial_metadata)
+
+            # Modify metadata (example: change file mode)
+            os.chmod(file_path, 0o755)
+            modified_metadata = os.stat(file_path)
+            log_metadata_operation(
+                "ModifyPermission", file_path, initial_metadata, modified_metadata
+            )
+
+            # Rename the file
+            new_file_path = file_path + "_renamed"
+            os.rename(file_path, new_file_path)
+            renamed_metadata = os.stat(new_file_path)
+            log_metadata_operation(
+                "Rename", new_file_path, initial_metadata, renamed_metadata
+            )
+
+            # Change group (example: change to the group 'users')
+            os.chown(new_file_path, -1, os.getgid())  # Set to the default group
+            group_changed_metadata = os.stat(new_file_path)
+            log_metadata_operation(
+                "ChangeGroup", new_file_path, renamed_metadata, group_changed_metadata
+            )
+
+            # Set extended attribute using setfattr
+            attr_name = "user.test_attr"
+            attr_value = "test_value"
+            set_extended_attribute(new_file_path, attr_name, attr_value)
+            retrieved_attr_value = get_extended_attribute(new_file_path, attr_name)
+            log_metadata_operation(
+                "Setxattr", new_file_path, None, f"{attr_name}={retrieved_attr_value}"
+            )
+
+            # Release the lock
+            fcntl.flock(lock_file, fcntl.LOCK_UN)
+
+    except Exception as e:
+        logging.error(f"Error during metadata operations: {str(e)}")
+
+
+if __name__ == "__main__":
```

Comment:
Would you like to run this page directly? if yes, why?

Reply:
This is a standalone script will run inside the fedora pods for few mins to generate load in cache.